lib/node_modules/@stdlib/ndarray/base/unary/test/test.2d.js

```diff
@@ -0,0 +1,1173 @@
+	var expected;
+	var bsize;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'float64';
+	ord = 'row-major';
+
+	bsize = blockSize( dt );
+	sh = [ 2, bsize*2 ];
+	st = [ bsize*4, -2 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, ones( numel( sh )*2, dt ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh )*2, dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Float64Array( x.length*2 );
+	dfill( x.length, 10.0, expected, st[ 1 ] );
+
+	t.strictEqual( isSameFloat64Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( x ) {
+		return x * 10.0;
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (row-major, contiguous, accessors)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'row-major';
+	sh = [ 2, 2 ];
+	st = shape2strides( sh, ord );
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, oneTo( numel( sh ), dt ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh ), dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array([
+		10.0,
+		0.0,
+		20.0,
+		0.0,
+		30.0,
+		0.0,
+		40.0,
+		0.0
+	]);
+	t.strictEqual( isSameComplex128Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( z ) {
+		return new Complex128( z.re*10.0, z.im*10.0 );
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (row-major, contiguous, negative strides, accessors)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'row-major';
+	sh = [ 2, 2 ];
+	st = [ -2, -1 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, oneTo( numel( sh ), dt ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh ), dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array([
+		10.0,
+		0.0,
+		20.0,
+		0.0,
+		30.0,
+		0.0,
+		40.0,
+		0.0
+	]);
+	t.strictEqual( isSameComplex128Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( z ) {
+		return new Complex128( z.re*10.0, z.im*10.0 );
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (row-major, non-contiguous, same sign strides, accessors)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'row-major';
+	sh = [ 2, 2 ];
+	st = [ 4, 1 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, oneTo( 8, dt ), sh, st, o, ord );
+	y = ndarray( dt, zeros( 8, dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array([
```

Comment:
This is a good example of where the use of `oneTo` for creating a complex number array is not great, as it makes it harder to determine which `0.0` is expected and which is not.